### DISCUSSION TITLE
Empty Config.json file

### DISCUSSION THREAD

**Discussion by armyturtle77** [qdm12/ddns-updater#793] [Q&A] [answered]
I am running CasaOS on top of Xubuntu. Casa has an "app store" where it lists this very useful utility. It looks rather easy to use, so I installed it. Unfortunately upon launching and going to my IP:8000  I get a blank page that has all the categories listed "Domain" "Host" "Provider" "IP version" "Update Status" "Set IP" and "Previous IPs reverse chronological order" 
But there's no information listed under them.
I found a config.json file under the folder  /DATA/AppData/ddns-updater/data/
and I'm not sure how to edit this properly, but when I open it with a text editor it's empty. I don't even know where to go to find the proper information I need to list in here, nor how the code needs to look. I'm using DynDNS. 
So in short:
1. What program should I open this file with to edit it
2. How can I put in my DynDNS information into this?

I've tried looking through the documentation here on Github and I just don't understand it properly.

**Comment by armyturtle77** [comment]
Okay, I found code here:  https://github.com/qdm12/ddns-updater/blob/master/config.json

But I still am not sure if I can just use gedit to copy/paste this into the config.json file. Also, I don't know what I need in here... I got a TSIG key from DynDNS and I tried putting my domain name and that key in the last entry areas... but it keeps telling me when I try to launch DDNS-updater that it's "Unhealthy"
Probably has something to do with me editing the config.json file with gedit and just saving it... maybe permissions issue? I don't know.
Help if you can please. :)

**Comment by qdm12** [comment]
https://github.com/qdm12/ddns-updater?tab=readme-ov-file#container

https://github.com/qdm12/ddns-updater?tab=readme-ov-file#configuration

https://github.com/qdm12/ddns-updater/blob/master/docs/dyndns.md

> but it keeps telling me when I try to launch DDNS-updater that it's "Unhealthy"

Share the ddns-updater container logs, otherwise there is no way to know what's wrong.

**Reply by armyturtle77** [reply]
Where can I find those logs?

**Reply by qdm12** [reply]
Isn't there a container logs section in the docker manager somewhere? Otherwise in a terminal, find the container id or name with `docker ps` and get the logs with `docker logs -f <ddns-updater-container-id-or-name>` (press ctrl+c to stop following logs)

**Reply by armyturtle77** [reply]
2024-08-25T15:49:53Z WARN cannot DNS resolve .turtles.dyndns.org after 5 tries: lookup .turtles.dyndns.org: no such host
2024-08-25T15:49:53Z INFO ipv4 address of .turtles.dyndns.org is invalid IP and your ipv4 address  is 69.204.163.6
2024-08-25T15:49:53Z INFO Updating record [domain: turtles.dyndns.org | host:  | provider: dyn | ip: ipv4] to use 69.204.163.6
2024-08-25T15:49:53Z ERROR hostname does not exist
2024-08-25T15:54:53Z INFO obtaining ipv4 address succeeded after 1 failed try
2024-08-25T15:54:53Z WARN cannot DNS resolve .turtles.dyndns.org after 5 tries: lookup .turtles.dyndns.org: no such host
2024-08-25T15:54:53Z INFO ipv4 address of .turtles.dyndns.org is invalid IP and your ipv4 address  is 69.204.163.6
2024-08-25T15:54:53Z INFO Updating record [domain: turtles.dyndns.org | host:  | provider: dyn | ip: ipv4] to use 69.204.163.6
2024-08-25T15:54:54Z ERROR hostname does not exist
2024-08-25T15:59:53Z WARN cannot DNS resolve .turtles.dyndns.org after 5 tries: lookup .turtles.dyndns.org: no such host
2024-08-25T15:59:53Z INFO ipv4 address of .turtles.dyndns.org is invalid IP and your ipv4 address  is 69.204.163.6
2024-08-25T15:59:53Z INFO Updating record [domain: turtles.dyndns.org | host:  | provider: dyn | ip: ipv4] to use 69.204.163.6
2024-08-25T15:59:53Z ERROR hostname does not exist
2024/08/25 16:03:52 "GET http://192.168.86.66:8000/ HTTP/1.1" from 192.168.86.87:49533 - 200 1602B in 49.425µs
2024/08/25 16:03:52 "GET http://192.168.86.66:8000/favicon.ico HTTP/1.1" from 192.168.86.87:49533 - 404 19B in 8.009µs
2024-08-25T16:04:53Z WARN cannot DNS resolve .turtles.dyndns.org after 5 tries: lookup .turtles.dyndns.org: no such host
2024-08-25T16:04:53Z INFO ipv4 address of .turtles.dyndns.org is invalid IP and your ipv4 address  is 69.204.163.6
2024-08-25T16:04:53Z INFO Updating record [domain: turtles.dyndns.org | host:  | provider: dyn | ip: ipv4] to use 69.204.163.6
2024-08-25T16:04:53Z ERROR hostname does not exist
2024-08-25T16:05:36Z WARN [http server] shutting down (context canceled)
2024-08-25T16:05:36Z WARN [healthcheck server] shutting down (context canceled)
2024-08-25T16:05:36Z WARN [http server] shut down
2024-08-25T16:05:36Z WARN [healthcheck server] shut down
2024-08-25T16:05:36Z WARN Caught OS signal, shutting down
2024-08-25T16:05:36Z INFO Shutdown successful

**Reply by armyturtle77** [reply]
See, I think the problem is that it keeps trying to insert a PERIOD (.) before whatever I type in for the domain name.

**Comment by armyturtle77** [comment]
While I wait to find out where those logs can be found... I was able to at least make a little progress. See the attached images for reference. I did "XXXXXX" out my user key for the purposes of sharing here online, but I have my actual key in there when running this.

Every time I run this now, it tries to insert a PERIOD (.) before my domain name, despite having "turtles.dyndns.org" in the code.  See how it makes it ".turltles.dyndns.org" in the UI once I run it?  
What am I doing wrong here?
![Screenshot 2024-08-23 140212](https://github.com/user-attachments/assets/876a6424-7c91-4295-97ff-6445c75534bd)
![Screenshot 2024-08-23 140118](https://github.com/user-attachments/assets/8867a444-0939-4fca-a556-394fd66a1d29)

**Reply by qdm12** [reply]
What version of the program are you using?
- if you use <= v2.7, you need to set domain to dyndns.org and owner to turtles (from the top of my head)
- if you use the latest program, set domain to `turtles dyndns.org` and remove the owner field.

**Reply by armyturtle77** [reply]
I'm using v 2.6.0 and no matter what I put in the domain field in the quotes... it ALWAYS inserts a PERIOD in front of the first character/letter I have in quotes.  Example:   "turtles.dyndns.org" BECOMES .turtles.dyndns.org   or  "dydns.org" BECOMES .dyndns.org

**Reply by qdm12** [reply, marked as answer]
If you use domain `"domain": "dyndns.org"` and owner/host `"host": "turtles"`, does it produce `.turtles.dydns.org`?

**Reply by armyturtle77** [reply]
So "host" is what worked. Without that it kept putting a (.) before whatever character was in the "domain": spot. Thanks for the help! I'm new to this and it's actually working!

**Reply by qdm12** [reply]
Yes the documentation is versioned, see the top of the readme. You were likely using the latest documentation which doesn't work with v2.6.0. Retro compatibility is maintained, but not forward compatibility (downgrading to a lower release).

**Reply by armyturtle77** [reply]
Yep! Now that you pointed that out to me I went back and found the documentation for 2.6.1 (which I'm assuming is the same for version 2.6.0, because that's what my version shows as). And there looks like there's no need for the "Owner": field for Dyndns in this one either. So I've removed that to clean up the code a bit. Thank you again for your help.

**Reply by qdm12** [reply]
`owner` was the name name for `host` only for a few days, before it got removed in favor of having everything in the domain field 😄  so I think for v2.6.x you would need to use `domain` and optionally `host` (defaults to `@` usually)
